src/timeout.ts

```diff
@@ -214,16 +215,18 @@ export class CSOTTimeoutContext extends TimeoutContext {
 
   get serverSelectionTimeout(): Timeout | null {
     // check for undefined
-    if (typeof this._serverSelectionTimeout !== 'object') {
+    if (typeof this._serverSelectionTimeout !== 'object' || this._serverSelectionTimeout?.cleared) {
+      const { remainingTimeMS, serverSelectionTimeoutMS } = this;
+      if (remainingTimeMS <= 0)
+        throw new MongoOperationTimeoutError('Timed out in server selection');
```

Comment:
Could we get a more descriptive error message here? Maybe including the serverSelectionTimeoutMS value and the time passed?

Reply:
Updated. Included the time that passed. I don't think the configured `serverSelectionTimeoutMS` value would be of use to a user when seeing this error.